### PR TITLE
adapt to saw-core-coq changes

### PR DESCRIPTION
**Pull request by Ptival**
This change is contingent upon the merge of: https://github.com/GaloisInc/saw-core/pull/192

Once the merge happens, we will need to update the saw-core submodule here for the change to make sense.

This commit allows for more flexible setups, in particular, before this, the `offline_coq` tactic did not have a convenient mean of specifying what extra imports should be included after the mandatory preamble.

### CODE REVIEW

File: src/SAWScript/Prover/Exporter.hs [hunk 1]
```diff
@@ -297,7 +316,10 @@ writeCoqTerm ::
   Term ->
   TopLevel ()
 writeCoqTerm name notations skips path t = do
-  let configuration = coqTranslationConfiguration notations skips
+  let configuration =
+        withImportSAWCorePrelude $
+        withImportCryptolPrimitivesForSAWCore $
```
Comment (RyanGlScott) [inline src/SAWScript/Prover/Exporter.hs:321]: As far as I can tell, these imports weren't required before this patch. Is there a reason that they are needed now?

File: src/SAWScript/Prover/Exporter.hs [hunk 2]
```diff
@@ -331,7 +353,10 @@ writeCoqCryptolModule inputFile outputFile notations skips = io $ do
   cryptolPrimitivesForSAWCoreModule <- scFindModule sc nameOfCryptolPrimitivesForSAWCoreModule
   (cm, _) <- loadCryptolModule sc env inputFile
   let cryptolPreludeDecls = map Coq.moduleDeclName (moduleDecls cryptolPrimitivesForSAWCoreModule)
-  let configuration = coqTranslationConfiguration notations skips
+  let configuration =
+        withImportSAWCorePrelude $
+        withImportCryptolPrimitivesForSAWCore $
```
Comment (RyanGlScott) [inline src/SAWScript/Prover/Exporter.hs:358]: A similar question here.

**Comment by brianhuffman** [comment]
#1176 (which has now been merged) updates the saw-core submodule to the revision immediately before GaloisInc/saw-core#192. So this PR is the next one that we need to merge in order to bump the saw-core submodule further.

File: src/SAWScript/Prover/Exporter.hs (hunk 1 above)
Comment (Ptival) [reply src/SAWScript/Prover/Exporter.hs:321]: The SAWCorePrelude used to be added via the extraPreamble.

The CryptolPrimitivesForSAWCore was not added, and is seemingly not needed when one run `write_coq_{...}` commands (at least I have not needed it), but is needed when running the `offline_coq` tactic from SAW script when it is working with Cryptol code.

I likely need to revisit exactly what imports are needed for what commands, and where the line is between imports that are mandatory and ones that are only needed when working with certain types of files. But if this PR is blocking others, I would be happy getting back to this decision in a subsequent one.

Comment (RyanGlScott) [reply src/SAWScript/Prover/Exporter.hs:321]: > The SAWCorePrelude used to be added via the extraPreamble.

True, but the `extraPremable` was only used in another function (`writeCoqCryptolPrimitivesForSAWCore`), not this function (`writeCoqTerm`), right?

> The CryptolPrimitivesForSAWCore was not added, and is seemingly not needed when one run `write_coq_{...}` commands (at least I have not needed it), but is needed when running the `offline_coq` tactic from SAW script when it is working with Cryptol code.

Ah, I think I assumed that this patch intended to simply update the `saw-core` submodule without affecting any functionality—I only read the commit message, not the PR description. But the PR description does indeed mention the change to `offline_coq`. In that case, I'll defer to your judgment.

**Review by RyanGlScott — APPROVED** [review]

**Comment by RyanGlScott** [comment]
I think the CI failure in `saw-remote-api-tests` is (at least partly) my fault. I've submitted #1178 in an attempt to repair the damage. Sorry for the delay!

**Pull request merged**